### PR TITLE
Encyption extension

### PR DESCRIPTION
**Pull request by mcardosos**
This migrates the managaed encypted vm disks sample to this repo.
The keyvault stuff is not working as expected yet :(

Thanks for your contribution! Please let us know what problems this PR addresses or what functionality it adds. Please link to other related issues and PRs to speed up review. Thank you!

### CODE REVIEW

**Comment by mcardosos** [comment]
Everything works!

**Comment by vladbarosan** [comment]
@mcardosos can you update the branch to latest ?

File: network/network.go [hunk 1]
```diff
@@ -86,7 +114,30 @@ func getSubnetsClient() network.SubnetsClient {
...
 
 // CreateVirtualNetworkSubnet creates a subnet
-func CreateVirtualNetworkSubnet() {}
+func CreateVirtualNetworkSubnet(ctx context.Context, vnetName, subnetName string) (subnet network.Subnet, err error) {
+	subnetsClient := getSubnetsClient()
+
+	future, err := subnetsClient.CreateOrUpdate(
+		ctx,
+		helpers.ResourceGroupName(),
+		vnetName,
```
Comment (vladbarosan) [inline network/network.go]: does the vnet already needs to exist when creating this ?

File: network/network.go [hunk 2]
```diff
@@ -27,6 +27,34 @@ func getVnetClient() network.VirtualNetworksClient {
...
+	}
+
+	err = future.WaitForCompletion(ctx, vnetClient.Client)
+	if err != nil {
+		return vnet, fmt.Errorf("cannot get the vnet create or update future response: %v", err)
+	}
+
+	return future.Result(vnetClient)
+}
+
```
Comment (vladbarosan) [inline network/network.go]: cant we just leave the `CreateVirtualNetworkAndSubnets` ? do we need this func also ?

File: compute/compute_test.go [hunk 3]
```diff
@@ -21,6 +22,9 @@ import (
 
 var (
 	vmName           = "az-samples-go-" + helpers.GetRandomLetterSequence(10)
+	accountName      = strings.ToLower("azuresamplesgo" + helpers.GetRandomLetterSequence(10))
+	vaultName        = "az-samples-go-" + helpers.GetRandomLetterSequence(10)
+	diskName         = "az-samples-go-" + helpers.GetRandomLetterSequence(10)
 	nicName          = "nic1"
 	username         = "az-samples-go-user"
 	password         = "NoSoupForYou1!"
```
Comment (vladbarosan) [inline compute/compute_test.go]: >"NoSoupForYou1!" [](start = 20, length = 16)

:D

File: compute/compute_test.go [hunk 4]
```diff
@@ -21,6 +22,9 @@ import (
 
 var (
 	vmName           = "az-samples-go-" + helpers.GetRandomLetterSequence(10)
+	accountName      = strings.ToLower("azuresamplesgo" + helpers.GetRandomLetterSequence(10))
+	vaultName        = "az-samples-go-" + helpers.GetRandomLetterSequence(10)
+	diskName         = "az-samples-go-" + helpers.GetRandomLetterSequence(10)
```
Comment (vladbarosan) [inline compute/compute_test.go]: what are these used for ?

File: compute/encryption.go [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+					},
+					DataDisks: &[]compute.DataDisk{
+						{
+							CreateOption: compute.DiskCreateOptionTypesAttach,
+							Lun:          to.Int32Ptr(0),
+							ManagedDisk: &compute.ManagedDiskParameters{
+								ID:                 disk.ID,
+								StorageAccountType: compute.StorageAccountTypes(storageAccount.Sku.Name),
+							},
+						},
```
Comment (vladbarosan) [inline compute/encryption.go]: Can we use managed disks ehre also ?

File: compute/compute_test.go (hunk 3 above)
Comment (mcardosos) [reply compute/compute_test.go]: Ah, yes that was @joshgav 's idea :D

File: compute/encryption_test.go [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+	helpers.PrintAndLog("created keyvault")
+
+	_, err = CreateManagedDisk(ctx, diskName)
+	if err != nil {
+		helpers.PrintAndLog(err.Error())
+	}
+	helpers.PrintAndLog("created disk")
+
+	_, err = network.CreateNetworkSecurityGroup(ctx, nsgName)
+	if err != nil {
```
Comment (vladbarosan) [inline compute/encryption_test.go]: do we need teh NSG ?

File: compute/compute_test.go (hunk 4 above)
Comment (mcardosos) [reply compute/compute_test.go]: To create more resources

File: network/network.go (hunk 1 above)
Comment (mcardosos) [reply network/network.go]: Yes

File: network/network.go (hunk 2 above)
Comment (mcardosos) [reply network/network.go]: We can leave the func that also creates the subnets

File: compute/encryption.go (hunk 5 above)
Comment (mcardosos) [reply compute/encryption.go]: Managed disks are already in use here when creating the VM. They were created in a previous step. Will update comment.

File: compute/encryption_test.go (hunk 6 above)
Comment (mcardosos) [reply compute/encryption_test.go]: Removed :D

**Comment by vladbarosan** [comment]
Ok LGTM, deffering this one to @marstr  for final approval as he worked on this previously

**Pull request merged**